wiki.moinmoin

```diff
@@ -91,216 +93,423 @@ Notes:
+RULE:   provide low level unit-tests. But more complex autopkgtests make no
+RULE:   sense on that level. Therefore in those cases one might want to test on
+RULE:   the solution level.
+RULE:   - Process wise MIR-requesting teams can ask (on the bug) for this
+RULE:     special case to apply for a given case, which reduces the test
+RULE:     constraints on the micro libraries but in return increases the
+RULE:     requirements for the test of the actual app/solution.
+RULE:   - Since this might promote micro-lib packages to main with less than
+RULE:     the common level of QA any further MIRed program using them will have
+RULE:     to provide the same amount of increased testing.
+TODO: - This package is minimal and will be tested in a more wide reaching
+TODO:   solution context <TBD>, details about this testing are here <TBD>
+
+[Quality assurance - packaging]
+RULE: - The package uses a debian/watch file whenever possible. In cases where
+RULE:   this is not possible (e.g. native packages), the package should either
+RULE:   provide a debian/README.source file or a debian/watch file (with
+RULE:   comments only) providing clear instructions on how to generate the
+RULE:   source tar file.
+TODO: - debian/watch is present and works
+TODO: - debian/watch is not present, instead it has <TBD>
+
+RULE: - It is often useful to run `lintian --pedantic` on the package to spot
+RULE:   the most common packaging issues in advance
+TODO: - This package does not yield massive lintian Warnings, Errors
+TODO: - Link to recent build log including a lintian run <TBD>
+
+RULE: - The package should not rely on obsolete or about to be demoted packages.
+RULE:   That currently includes package dependencies on Python2 (without
+RULE:   providing Python3 packages), and packages depending on GTK2.
+TODO: - This package does not rely on obsolete or about to be demoted packages.
+TODO: - This package has no python2 or GTK2 dependencies
+
+RULE: - Debconf questions should not bother the default user too much
+TODO: - The package will be installed by default, but does not ask debconf
+TODO:   questions higher than medium
+TODO: - The package will not be installed by default
+
+RULE  - The source should be reasonably easy to understand and maintain.
+TODO: - Packaging and build is easy, link to d/rules <TBD>
+TODO: - Packaging is complex, but that is ok because <TBD>
 
 [UI standards]
+TODO: - Application is not end-user facing (does not need translation)
+TODO: - Application is end-user facing, Translation is present, via standard
+TODO:   intltool/gettext or similar build and runtime internationalization
+TODO:   system see <TBD>
+TODO: - End-user applications that ships a standard conformant desktop file,
+TODO:    see <TBD>
+TODO: - End-user applications without desktop file, not needed because <TBD>
 
 [Dependencies]
+RULE: - In case of alternative the preferred alternative must be in main.
+RULE: - Build(-only) dependencies can be in universe
```

Comment:
See my comment above for statically linked code (basically, when the build-dep code ends up being in the binary)